### PR TITLE
cansniffer: add CAN FD support

### PR DESCRIPTION
**Pull request by hartkopp**

### CODE REVIEW

File: cansniffer.c [hunk 1]
```diff
@@ -343,6 +365,23 @@ int main(int argc, char **argv)
 		}
 	}
 
+	/* enable CAN FD if not disabled by comand line option */
+	if (canfd_mode != CANFD_OFF) {
+
```
Comment (marckleinebudde) [inline cansniffer.c:370]: nitpick: remove empty line

File: cansniffer.c [hunk 2]
```diff
@@ -560,18 +606,39 @@ int handle_frame(int fd, long currcms)
...
 
+	/* CAN FD auto mode: switch based on first reception */
+	if (canfd_mode == CANFD_AUTO) {
+		if (nbytes == CAN_MTU) {
+			canfd_mode = CANFD_OFF;
+			/* change back auto defaults for Classical CAN */
+			max_dlen = CAN_MAX_DLEN;
+			name_sep = CC_SEP;
+			print_ascii = 1;
+		} else
```
Comment (marckleinebudde) [inline cansniffer.c:630]: nitpick `{` `}` on both side of `else`

**Review by marckleinebudde — CHANGES_REQUESTED** [review]
just some nitpicks

File: cansniffer.c (hunk 2 above)
Comment (hartkopp) [reply cansniffer.c]: fixed and force pushed ;-)

**Pull request merged**